api/routes/utils.js

```diff
@@ -538,25 +545,21 @@ var utils = {
 
           // Reset the percentage
           investment.percentage = 100 * investment.amount/expertCategoryTotal;
-
-          // Reset the valuation
-          investment.valuation = Math.floor(investment.percentage/100 * expertCategoryTotal);
-          investor.portfolio[j].investments[q] = investment;
         }
       }
     }
     return investors;
   },
 
-  // Given a revoke that just happened, update roi
-  updateROI: function(oldROI, roiFromRevoke) {
-    var newLen = oldROI.length + 1;
-    var newVal = (((oldROI.value * oldROI.length + roiFromRevoke)/(newLen)).toFixed(2))/1;
+  // Given a revoke that just happened, update dividends
```

Comment:
I think that updating the dividends will be a lot different than updating the ROI.

ROI is specific to an investor. My ROI has nothing to do with yours. Dividends are tied together. When you invest more in my favorite expert, my dividends go up. Dividends need to be recalculated every time an expert's reps change.

That being said, the calculation is extremely simple. It's just an investment's percentage * total * 0.1.

I think we may be able to avoid putting this in the model at all. We already have the percentage and the total fairly easily. If we want to calculate and store it, then it will have to be updated for all users every time an expert's total changes. This will be similar to the way we update all of the investor's percentiles every time an investment changes.